src/lib/events.ts

```diff
@@ -0,0 +1,59 @@
+import {ITaskService, ActiveTaskService} from "./service";
+import EventEmitter from "events";
+import {Logger} from "winston";
+
+export type TaskEvent = LaunchTaskEvent | PendingTaskEvent | ConsoleTaskEvent | HealthTaskEvent | ErrorTaskEvent | ExitTaskEvent;
+
+export interface PendingTaskEvent {
```

Comment:
as part of this PR, IMO we should very clearly state in typedoc form what these interfaces represent.

Reply:
further, is it possible / does it make sense to have these extend from a base `Event` interface? perhaps:

```typescript
export interface TaskEvent {
  name: string;
  service: ITaskService;
  logger: Logger;
  data: any; // can/should improve this of course.
}
```